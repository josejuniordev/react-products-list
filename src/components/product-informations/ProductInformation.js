import React from 'react';
import './ProductInformation.scss';
import ImageDisplay from '../product-parts/image-display/ImageDisplay';
import ContentHeader from '../layout/content-header/ContentHeader';
import { componentsSizes } from '../../constants/components-sizes';

function ProductInformation(
  {
    product,
  },
) {
  return (
    <article className="product-information">
      <section className="product-information__image-and-details">
        <div className="product-information__image">
          <ImageDisplay product={product} />
        </div>
        <div className="product-information__details">
          <h2 className="product-information__title">Detalhes do produto</h2>
          <p>{product.longDescription}</p>
        </div>
      </section>
      <section className="product-information__datasheet">
        <ContentHeader size={componentsSizes.SMALL} title="Ficha tÃ©cnica" />
        <ul className="product-information__datasheet-list">
          {
            product.datasheet && product.datasheet.length
              && product.datasheet.map((item, index) => (
                <li key={index}>
                  <span>
                    {item.titulo}
:
                  </span>
                  {' '}
                  {item.descricao}
                </li>
              ))
          }
        </ul>
      </section>
    </article>
  );
}

export default ProductInformation;
