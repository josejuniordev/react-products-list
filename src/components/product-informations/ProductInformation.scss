@import "../../styles/colors";
@import "../../styles/bem";

.product-information {

  @include element(image-and-details) {
    display: flex;
    margin-bottom: 50px;

    @media (max-width: 425px) {
      flex-direction: column;
    }
  }

  @include element(image) {
    width: 360px;
    height: 360px;
    margin-right: 50px;
    flex-shrink: 0;

    @media (max-width: 425px) {
      width: 100%;
      margin: 0 0 30px 0;
    }
  }

  @include element(title) {
    font-size: 20px;
    line-height: 33px;
    margin: 0 0 25px 0;
    color: $text-color;
  }

  @include element(details) {
    color: $secondary-text-color;
  }

  @include element(datasheet-list) {
    color: $secondary-text-color;
    -webkit-columns: 300px 3;
    -moz-columns: 300px 3;
    columns: 300px 3;
    column-gap: 3em;
  }
}
