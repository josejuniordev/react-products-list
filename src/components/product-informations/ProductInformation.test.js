import { shallow } from 'enzyme';
import React from 'react';
import ProductInformation from './ProductInformation';
import { productsMock } from '../../mock/products-mock';

describe('<ProductInformation />', () => {
  const rendered = shallow(<ProductInformation product={productsMock[0]} />);

  it('should render correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
