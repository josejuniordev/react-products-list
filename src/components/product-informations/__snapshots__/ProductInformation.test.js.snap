// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInformation /> should render correctly 1`] = `ShallowWrapper {}`;
